frontend/src/stores/knowledges/actions.ts

```diff
@@ -0,0 +1,27 @@
+import { get, post } from '@/services/http'
+import { getKnowledge } from './getters'
+import { knowledges } from './state'
+import type { Knowledge } from '@/domain/Knowledge'
+import type { Role } from '@/domain/Role'
+
+async function fetchKnowledgesByRoleId(roleId: number): Promise<Knowledge[]> {
+  const roleResponse = await get<Role>(`/roles/${roleId}`)
```

Comment:
Remember: always check first if you have in state. If you don't have it in the state, then do the request to the backend